lib/mixlib/install/backend/artifactory.rb

```diff
@@ -19,72 +19,114 @@
-            raise e, "unstable channel uses endpoint #{ARTIFACTORY_ENDPOINT} \
-which is currently only accessible through Chef's internal network."
-          end
+          artifacts = artifactory_info.collect { |a| create_artifact(a) }
+          artifacts_for_version = artifacts.find_all { |a| a.version == resolve_version(artifacts) }
 
           if options.platform
-            artifact(results.first)
-          else
-            results.collect do |result|
-              artifact(result)
+            artifacts_for_version.find do |a|
+              a.platform == options.platform &&
+                a.platform_version == options.platform_version &&
+                a.architecture == options.architecture
             end
+          else
+            artifacts_for_version
           end
         end
 
-        private
+        # Fetches all artifacts from the configured Artifactory repository using
+        # channel and product name as search criteria
+        #
+        # @return [Array<Hash>] list of artifactory hash data
+        #
+        # Hash data:
+        #   download_uri: The full url download path
+        #   <property_name>: The names of the properties associcated to the artifact
+        #
+        def artifactory_info
+          query = <<-QUERY
+items.find(
+  {"repo": "omnibus-#{options.channel}-local"},
+  {"@omnibus.project": "#{options.product_name}"}
+).include("name", "path", "repo", "property")
+          QUERY
 
-        def artifact(result)
-          ArtifactInfo.new(
-            md5:              result["properties"]["omnibus.md5"].first,
-            sha256:           result["properties"]["omnibus.sha256"].first,
-            version:          result["properties"]["omnibus.version"].first,
-            platform:         result["properties"]["omnibus.platform"].first,
-            platform_version: result["properties"]["omnibus.platform_version"].first,
-            architecture:     result["properties"]["omnibus.architecture"].first,
-            url:              result["downloadUri"]
+          # Creates a client based on ENVs:
+          #   ARTIFACTORY_ENDPOINT
+          #   ARTIFACTORY_USERNAME
+          #   ARTIFACTORY_PASSWORD
+          client = ::Artifactory::Client.new(
+            endpoint: ENV["ARTIFACTORY_ENDPOINT"] || "http://artifactory.chef.co"
           )
-        end
 
-        def params
-          params = {
-            "repos" => ARTIFACTORY_REPOSITORY,
-            "omnibus.version" => options.product_version
-          }
+          results = client.post("#{::Artifactory.endpoint}/api/search/aql",
+                                query, "Content-Type" => "text/plain")
 
-          if options.platform
-            params["omnibus.platform"] = options.platform
-            params["omnibus.platform_version"] = options.platform_version
-            params["omnibus.architecture"] = options.architecture
+          results["results"].collect do |result|
```

Comment:
Will do.  I'l going to spend a little time today to see if I can have AQL return the full downloadUri in the results.